src/Chapter3.hs

```diff
@@ -50,6 +50,7 @@ signatures in places where you can't by default. We believe it's helpful to
 provide more top-level type signatures, especially when learning Haskell.
 -}
 {-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
```

Comment:
Interesting action! But looking at this line I already can imagine how you solved the advanced task :smile: But let's see!